lib/bundler/definition.rb

```diff
@@ -358,10 +358,20 @@ def ensure_equivalent_gemfile_and_lockfile(explicit_flag = false)
       changed = []
 
       gemfile_sources = sources.lock_sources
-      if @locked_sources != gemfile_sources
-        new_sources = gemfile_sources - @locked_sources
-        deleted_sources = @locked_sources - gemfile_sources
 
+      new_sources = gemfile_sources - @locked_sources
+      deleted_sources = @locked_sources - gemfile_sources
+
+      new_deps = @dependencies - @locked_deps
+      deleted_deps = @locked_deps - @dependencies
+
+      # Check if it is possible that the source is only changed thing
+      if (new_deps.empty? && deleted_deps.empty?) && (new_sources.any? && deleted_sources.any?)
+        new_sources.reject! {|source| source.instance_of?(Bundler::Source::Path) && source.path.exist? }
```

Comment:
Actually, I think is_a? won't work because there's a Git subclass of Path and we don't want that got handled.